cmd/kubectl_token.go

```diff
@@ -0,0 +1,546 @@
+package cmd
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"math/big"
+	"net/http"
+	url2 "net/url"
+	"os"
+	"os/exec"
+	"os/signal"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/rancher/norman/types/convert"
+	managementClient "github.com/rancher/types/client/management/v3"
+	"github.com/urfave/cli"
+	"golang.org/x/crypto/ssh/terminal"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/pkg/apis/clientauthentication/v1beta1"
+)
+
+type LoginInput struct {
+	server       string
+	userID       string
+	clusterID    string
+	authProvider string
+	caCerts      string
+	skipVerify   bool
+}
+
+const (
+	kubeConfigCache = "/.cache/token"
+	cachedFileExt   = ".json"
+	authProviderURL = "%s/v3-public/authProviders"
+	authTokenURL    = "%sv3-public/authtokens/%s"
```

Comment:
Should the url be "authTokens" camelCase just like "authProviders"? or should it be non-camelCase - depends on what the pattern is so far for CLI, but needs to be consistent